ansible/roles/test/files/ptftests/advanced-reboot.py

```diff
@@ -279,24 +287,88 @@ def generate_arp_responder_conf(self, vlan_host_map):
 
     def dump_arp_responder_config(self, dump):
         # save data for arp_replay process
-        with open("/tmp/from_t1.json", "w") as fp:
+        filename = "/tmp/from_t1.json" if self.preboot_type is None else "/tmp/from_t1_%s.json" % self.preboot_type
+        with open(filename, "w") as fp:
             json.dump(dump, fp)
 
+    def get_peer_dev_info(self):
+        content = self.read_json('peer_dev_info')
+        for key in content.keys():
+            if 'ARISTA' in key:
+                self.vm_dut_map[key] = dict()
+                self.vm_dut_map[key]['mgmt_addr'] = content[key]['mgmt_addr']
+
+    def get_portchannel_info(self):
+        content = self.read_json('portchannel_ports_file')
+        for key in content.keys():
+            for member in content[key]['members']:
+                for vm_key in self.vm_dut_map.keys():
+                    if member == self.vm_dut_map[vm_key]['dut_port']:
+                        self.vm_dut_map[vm_key]['dut_portchannel'] = key
+                        self.vm_dut_map[vm_key]['neigh_portchannel'] = 'Port-Channel 1'
+                        break
+
+    def get_neigh_port_info(self):
+        content = self.read_json('neigh_port_info')
+        for key in content.keys():
+            if content[key]['name'] in self.vm_dut_map.keys():
+                self.vm_dut_map[content[key]['name']]['dut_port'] = key
+                self.vm_dut_map[content[key]['name']]['neigh_port'] = content[key]['port']
+                self.vm_dut_map[content[key]['name']]['ptf_port'] = self.port_indices[key]
+
+    def build_peer_mapping(self):
+        '''
+            Builds a map of the form
+                    'ARISTA01T1': {'mgmt_addr':
+                                   'neigh_portchannel'
+                                   'dut_portchannel'
+                                   'neigh_port'
+                                   'dut_port'
+                                   'ptf_port'
+                                    }
+        '''
+        self.vm_dut_map = {}
+        for file in self.test_params['preboot_files'].split(','):
+            self.test_params[file] = '/tmp/' + file + '.json'
+        self.get_peer_dev_info()
+        self.get_neigh_port_info()
+        self.get_portchannel_info()
+
     def setUp(self):
+        self.fails['dut'] = set()
         self.port_indices = self.read_port_indices()
         self.portchannel_ports = self.read_portchannel_ports()
         self.vlan_ports = self.read_vlan_ports()
+        if self.test_params['preboot_type'] is not None:
```

Comment:
preboot_type is too close to reboot_type, can you rename this variable to something more distinctive?

Reply:
preboot_oper?

Reply:
Done